pkg/agent/openflow/pipeline.go

```diff
@@ -2615,19 +2613,19 @@ func (c *client) createOFTable(table binding.Table, nextID uint8, missAction bin
 
 // igmpPktInFlows sets reg0[28] to mark the IGMP packet in MulticastTable and sends it to antrea-agent on MulticastTable.
 func (c *client) igmpPktInFlows(reason uint8) []binding.Flow {
-	var flows []binding.Flow
-	for _, dstIP := range []net.IP{mcastAllRouters, igmpV3ReportDst} {
+	flows := []binding.Flow{
 		// Set a custom reason for the IGMP packets, and then send it to antrea-agent and forward it normally in the
 		// OVS bridge, so that the OVS multicast db cache can be updated, and antrea-agent can identify the local multicast
 		// group and its members in the meanwhile.
-		flows = append(flows, MulticastTable.BuildFlow(priorityHigh).
+		// Do not set dst IP address because IGMPv1 report message uses target multicast group as IP destination in
+		// the packet.
+		MulticastTable.BuildFlow(priorityHigh).
 			MatchProtocol(binding.ProtocolIGMP).
 			MatchRegMark(FromLocalRegMark).
-			MatchDstIP(dstIP).
```

Comment:
Should we add `MatchDstIPNet(*mcastCIDR).`

Reply:
Not necessary, we already use IGMP protocol as a valid filter, the flow is working on IGMP messages even if not setting the IP destination. As for other unitcast traffic, it should not hit IGMP protocol.